src/chrome/internal/sources/identifiedLoadedSource.ts

```diff
@@ -0,0 +1,78 @@
+import * as fs from 'fs';
+import { IScript } from '../scripts/script';
+import { IResourceIdentifier } from './resourceIdentifier';
+import { ILoadedSource, IScriptMapper, ICurrentScriptRelationshipsProvider as IScriptMapperProvider, ContentsLocation, SourceScriptRelationship, ImplementsLoadedSource, ScriptAndSourceMapper } from './loadedSource';
+import { ILoadedSourceToScriptRelationship } from './loadedSourceToScriptRelationship';
+import _ = require('lodash');
+import { printArray } from '../../collections/printing';
+import { LocationInScript, LocationInLoadedSource } from '../locations/location';
+
+/**
+ * Loaded Source classification:
+ * Is the script content available on a single place, or two places? (e.g.: You can find similar scripts in multiple different paths)
+ *  1. Single: Is the single place on storage, or is this a dynamic script?
+ *      Single path on storage: RuntimeScriptRunFromStorage
```

Comment:
Let's talk about this, I can't tell what happens in the case where there is a `/foo.js` script in the runtime, and also a file at that path on disk, but the script in the runtime is dynamic and has different contents than the file on disk which is actually the source. This is anything that transpiles in memory like ts-node.

Reply:
Also there is a difference between "Dynamic" scripts that have paths and ones that don't

Reply:
As discussed offline, the current code won't support that case nor ts-node. We can implement it in a future version.